### PR TITLE
Redesign iOS audiobook detail page

### PR DESCRIPTION
**Pull request by Quick104**
## What & why

The iOS/macOS audiobook detail page didn't follow the app's own design language (`DESIGN_LANGUAGE.md` §5.5). Movies get the blurred-art hero + inverted play pill; audiobooks got a centered square on flat black with author, all narrators, and runtime jammed into one centered run-on line. This redesign brings audiobooks up to the same bar and adds the audiobook-specific affordances (narrator hierarchy, chapters, progress, speed).

Design research + an HTML mockup of the direction live in `docs/audiobook-detail-redesign/mockup.html`.

## Changes (iOS/macOS — tvOS layout preserved verbatim)

- **Ambient hero** — the square cover floats on a blurred, color-extracted wash of itself that bleeds under the status bar (the ScrollView ignores the top safe area; the backdrop is a top-pinned background and the cover is inset past it).
- **Clear hierarchy** — cleaned Title → Series line → Author → Narrator, replacing the single run-on metadata string.
- **Progress-forward CTA** — white Resume pill with an in-pill progress bar + "% complete / time left" for in-progress books; falls back to Play / Play Again. Secondary controls: Start Over, Speed (wired to `AudioPlayerViewModel.setPlaybackRate`), and Narration (when alternates exist).
- **Chapters first-class** (collapsible); **Parts** only renders for genuine multi-file books; codec/container demoted to a quiet details line.

## Bug fixes

- **`0 kbps`** — dropped bitrate from the part subtitle (it was integer-dividing a kbps value by 1000).
- **"Book 5 of 19"** — the series grouping includes alternate editions, so the total now comes from the title's `(N of M)` locator, not the entry count.
- **Narrator pile-up** — the server returns the cast as one comma-joined string, so the summariser splits on commas before rolling up to "& N more".

## Notes for reviewers

- New `AudiobookDetailFormatting` holds the pure title-cleanup / credit-summary logic, covered by `AudiobookDetailFormattingTests` (regression cases for the series-total and combined-narrator-string bugs).
- New Swift files are picked up via `project.yml` globbing — run `xcodegen generate` before building.
- **Intentionally deferred:** the mockup's *Download* and *Mark finished* controls have no client/server backing yet, so they were left out rather than shipped as dead buttons.

## Verification

- `Silo` (iOS) and `SiloTV` (tvOS) builds succeed.
- Formatter logic verified (the in-sim `.xctest` bundle wouldn't load in this environment, so the pure logic was checked out-of-band; the XCTest cases are included).
- Confirmed rendering in the iOS simulator.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **New Features**
  - Redesigned the audiobook detail screen with an updated hero layout, improved playback controls (including an in-pill resume with progress), richer chapters/series rail sections, and clearer audio metadata.
- **Bug Fixes**
  - Cleaned up audiobook titles and series labels by removing redundant series prefixes and trailing volume markers.
  - Improved narrator/credit summarization and standardized part/section subtitle formatting, removing technical bitrate-style details.
- **Tests**
  - Added coverage to ensure consistent title, series, and credit formatting across edge cases.

### CODE REVIEW

File: docs/audiobook-detail-redesign/mockup.html [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      <button class="abk-resume">
+        <i class="ti ti-player-play"></i>Resume
+        <div class="ptrack"><div class="pfill"></div></div>
+      </button>
+      <div class="abk-secondary">
+        <div class="abk-sbtn"><div class="abk-circ"><i class="ti ti-refresh"></i></div><small>Start over</small></div>
+        <div class="abk-sbtn"><div class="abk-circ"><i class="ti ti-download"></i></div><small>Download</small></div>
+        <div class="abk-sbtn"><div class="abk-circ speed">1.0×</div><small>Speed</small></div>
+        <div class="abk-sbtn"><div class="abk-circ"><i class="ti ti-check"></i></div><small>Finished</small></div>
+      </div>
```
[the remark this answers is not in the document]
Comment (Quick104) [reply docs/audiobook-detail-redesign/mockup.html:173]: Removed the deferred Download and Finished controls from the mockup so it matches the shipped action row. Narration is conditional (only renders when a title has alternate narrations), and this example book has none, so it's intentionally not shown here.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: iosApp/iosApp/Screens/Detail/AudiobookDetailContent.swift [hunk 2]
```diff
@@ -455,6 +845,40 @@ struct AudiobookDetailContent: View {
...
+        guard let position = detail.userData?.positionSeconds,
+              position.isFinite,
+              position > 30 else { return nil }
+        if let duration = detail.userData?.durationSeconds,
+           duration.isFinite,
+           duration > 0,
+           position >= duration - 5 {
+            return nil
+        }
+        return position
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Screens/Detail/AudiobookDetailContent.swift:857]: Fixed in 8d190bf — `resumePosition` now uses `totalDurationSeconds` for the near-end check, so a completed book with missing/stale `userData.durationSeconds` still suppresses Resume and falls back to Play Again.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: iosApp/iosApp/Screens/Detail/AudiobookDetailContent.swift [hunk 3]
```diff
@@ -470,45 +894,36 @@ struct AudiobookDetailContent: View {
...
-        }
-        return position
-    }
-
     private var primaryPlayLabel: String {
         if let resumePosition {
             return "Resume \(PlayerTimeFormatter.formatHMS(resumePosition))"
         }
         return "Play"
     }
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Screens/Detail/AudiobookDetailContent.swift:938]: Fixed in 8d190bf — hoisted `isFinished` / `primaryAction()` / `primaryActionIcon` to shared scope so the tvOS hero now shows "Play Again" and routes completed books through the restart path, matching iOS/mac.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
